.github/workflows/typescript-type-generation.yml

```diff
@@ -3,18 +3,76 @@ name: TypeScript Type Generation
 on:
   pull_request:
     paths:
-      - 'openmetadata-spec/src/main/resources/json/schema/**'
+      - "openmetadata-spec/src/main/resources/json/schema/**"
+      - "openmetadata-ui/src/main/resources/ui/src/generated/**"
   workflow_dispatch:
     inputs:
       branch:
-        description: 'Branch to run the workflow on'
+        description: "Branch to run the workflow on"
         required: true
-        default: 'main'
+        default: "main"
         type: string
 
+concurrency:
+  group: typescript-generation-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
 jobs:
   generate-types:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
     steps:
-      - name: Placeholder
-        run: echo "Workflow triggered successfully on branch ${{ github.event.inputs.branch || github.ref_name }}"
\ No newline at end of file
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch || github.event.pull_request.head.ref || github.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Switch to branch (if manual trigger)
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git checkout ${{ github.event.inputs.branch }}
```

Comment:
This should not be required! `actions/checkout` step will take care of cloning the repo with right branch!